internal/rbd/rbd_util.go

```diff
@@ -883,6 +869,38 @@ func genVolFromVolID(ctx context.Context, volumeID string, cr *util.Credentials,
 	return rbdVol, err
 }
 
+// genVolFromVolID generates a rbdVolume structure from the provided identifier, updating
+// the structure with elements from on-disk image metadata as well.
+func genVolFromVolID(ctx context.Context, volumeID string, cr *util.Credentials, secrets map[string]string) (*rbdVolume, error) {
+	vol, err := generateVolumeFromVolumeID(ctx, volumeID, cr, secrets)
+	if !errors.Is(err, util.ErrKeyNotFound) && !errors.Is(err, util.ErrPoolNotFound) {
+		return vol, err
+	}
+
+	// If the volume details are not found in the OMAP it can be a mirrored RBD
+	// image and the OMAP is already generated and the volumeHandle might not
+	// be the same in the PV.Spec.CSI.VolumeHandle. Check the PV annotation for
+	// the new volumeHandle. If the new volumeHandle is found, generate the RBD
+	// volume structure from the new volumeHandle.
+	c := util.NewK8sClient()
+	listOpt := metav1.ListOptions{
+		LabelSelector: PVReplicatedLabelKey,
+	}
+	pvlist, pErr := c.CoreV1().PersistentVolumes().List(context.TODO(), listOpt)
```

Comment:
This will be fetched only if the omap is not found and we are fetching based on the selector. as this is a temp solution i don't see a problem with it.